Mobile data viewer

diksipav opened the pull request.

jaclarke commented on shared/common/ui/select/index.tsx:
```diff
@@ -179,19 +189,41 @@ export function Select<T extends any>({
+              </button>
+            )}
+            {!!searchable &&
+              (fullScreen ? (
```
I think this should be `fullScreen && isMobile`

jaclarke replied:
Actually, there's a few places where you use just `fullScreen` that I think should also be `fullScreen && isMobile`. Maybe change the `const isMobile = useIsMobile();` to something like `const isFullscreenMobile = useIsMobile() && fullscreen;`?

jaclarke commented on shared/common/ui/select/index.tsx:
```diff
@@ -201,6 +233,7 @@ export function Select<T extends any>({
                         [styles.dropdownItemSelected]:
                           dropdown.selectedItemId === result.obj.item.item.id,
                         [styles.disabled]: !!result.obj.item.item.disabled,
+                        [styles.fullScreen]: !!fullScreen,
```
Rather than setting this class on every item, I would just have the one on the dropdown element.

diksipav replied:
I will move font styles upper, the padding and border is needed per item though.

jaclarke commented:
Nested objects don't render correctly in the fullscreen dropdown: 
![image](https://github.com/edgedb/edgedb-ui/assets/2085287/5191acfd-c451-4b05-8f20-60e3ca891026)
![image](https://github.com/edgedb/edgedb-ui/assets/2085287/dd04fe14-5e32-4448-9980-571841abed4c)

jaclarke commented:
Not full width when you're in a nested view:
![image](https://github.com/edgedb/edgedb-ui/assets/2085287/414df0bd-00b2-4cf2-b4c9-e90dee75a38b)

jaclarke commented:
If there's an error in the filter expression, clicking apply closes the popup without any indication of an error.

diksipav commented:
> If there's an error in the filter expression, clicking apply closes the popup without any indication of an error.
> 
> I saw this today when testing, I fixed this and also updated filterError state to also save the filter string, and to disable "APLLY" when there's an error and user didn't update the filter str.

diksipav commented:
> If there's an error in the filter expression, clicking apply closes the popup without any indication of an error.


I saw this today when testing, I fixed this and also updated filterError state to also save the filter string, and to disable "APLLY" when there's an error and user didn't update the filter str.

diksipav commented:
> Not full width when you're in a nested view: ![image](https://user-images.githubusercontent.com/2085287/266652947-414df0bd-00b2-4cf2-b4c9-e90dee75a38b.png)

Hey, this wasn't the full PR, nested views were not implemented. Now should works fine. : )

jaclarke commented on shared/studio/tabs/dataview/state/index.ts:
```diff
@@ -267,9 +269,10 @@ export class DataInspector extends Model({
 
   filter: prop<string>(""),
-  filterError: prop<string>(""),
+  errorFilter: prop<ErrorFilter | null>(null),
```
Not really a issue, but why flip the words? To me `filterError` sounds more 'correct'.

diksipav replied:
because I also save the filter : ) 
`filterError` is more like the error string, and `errorFilter` is more like filter type

jaclarke commented on shared/studio/tabs/dataview/dataInspector.tsx:
```diff
@@ -951,3 +1002,80 @@ const ExpandedDataInspector = observer(function ExpandedDataInspector({
+        fields.map((field, index) => {
+          const isMulti = field.multi;
+          const data = item.data;
+          const value = isMulti ? data[field.name].length : data[field.name];
```
`isMulti` is the wrong thing to check here to differentiate between props and links (links can be single and props can be multi). Also in the query links are fetched with `limit 10`, so you need to display the `__count_*` field it fetches instead.

```suggestion
          const value = field.type === ObjectFieldType.link ? data[`__count_${field.name}`] : data[field.name];
```

jaclarke commented on shared/studio/tabs/dataview/dataInspector.tsx:
```diff
@@ -951,3 +1002,80 @@ const ExpandedDataInspector = observer(function ExpandedDataInspector({
+                  {`${isMulti ? "multi" : ""} ${field.typename}`}
+                </span>
+              </div>
+              {isMulti ? (
```
Same as above.

jaclarke commented on shared/studio/tabs/dataview/dataInspector.tsx:
```diff
@@ -951,3 +1002,80 @@ const ExpandedDataInspector = observer(function ExpandedDataInspector({
+                    );
+                  }}
+                >
+                  {field.typename.split("::")[1]}
```
This doesn't work for types in nested modules. You need to get the last part:
```suggestion
                  {field.typename.split("::").pop()}
```

jaclarke commented:
Styling bugs:
- Mobile expanded view doesn't scroll
![image](https://github.com/edgedb/edgedb-ui/assets/2085287/1503786b-2b19-4dcd-9fd3-2d4831b69ef4)
- Link button should be monospace font
![image](https://github.com/edgedb/edgedb-ui/assets/2085287/7667c1f6-517e-47c3-a33b-c93a088b48e0)
- Divider lines
![image](https://github.com/edgedb/edgedb-ui/assets/2085287/5c3de4b3-fecb-4b01-bfaa-3365c3b226ff)
- Gap on desktop nested view
![image](https://github.com/edgedb/edgedb-ui/assets/2085287/bd802bbb-ac06-4103-b2f7-10b3a12a8866)

jaclarke commented:
More of a design issue, but filter is missing on nested views. The previous version where the back arrow, path and filter were at the bottom seemed to make more sense to me. The header at the top seems confusing as it matches the expanded popup view.

![image](https://github.com/edgedb/edgedb-ui/assets/2085287/a67cb336-ef93-4fbb-bda7-f839fb00a266)

jaclarke commented on shared/studio/tabs/dataview/dataInspector.tsx:
```diff
@@ -937,7 +990,9 @@ const ExpandedDataInspector = observer(function ExpandedDataInspector({
                   rowData.state.objectId,
                   rowData.state.objectTypeName,
-                  state.fields!.find((field) => field.name === item.fieldName)!
+                  state.mobileFields!.find(
+                    (field) => field.name === item.fieldName
+                  )!
```
I don't think this change is needed. It's doing a lookup by name, so it doesn't matter if it's using the full array of fields.

jaclarke commented:
I didn't notice before, but in the expanded mobile inspector we aren't showing all the fields on subtypes. I think we can leave this to a follow up PR though.
